platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormTest.java

```diff
@@ -8,32 +8,39 @@
 import com.softicar.platform.emf.EmfMarker;
 import com.softicar.platform.emf.action.marker.EmfCommonActionMarker;
 import com.softicar.platform.emf.editor.EmfEditAction;
-import com.softicar.platform.emf.test.EmfTestSubObject;
+import com.softicar.platform.emf.form.scope.EmfFormViewScopeAction;
 import com.softicar.platform.emf.test.simple.EmfTestObject;
+import com.softicar.platform.emf.test.simple.scoped.EmfScopedTestObject;
 import org.junit.Test;
 
-public class EmfFormTest extends AbstractEmfFormTest {
+public class EmfFormTest extends AbstractEmfFormTest<EmfScopedTestObject> {
 
 	private static final Day SOME_DAY = Day.fromYMD(2019, 4, 3);
 	private static final Day OTHER_DAY = Day.fromYMD(2019, 6, 7);
 	private static final String SOME_NAME = "Foo";
 	private static final String OTHER_NAME = "Bar";
+	private final EmfTestObject scopeObject;
+
+	public EmfFormTest() {
+
+		this.scopeObject = insertTestObject("Scope");
+	}
 
 	@Test
 	public void testCreationOfNewEntity() {
 
 		// create entity and show form
-		EmfTestSubObject entity = showForm(new EmfTestSubObject());
+		EmfScopedTestObject entity = showForm(createScopedTestObject());
```

Comment:
Okay, we now test scoped objects - but we no longer test sub-objects (that is, `implements IEmfSubObject`... that is, objects with a `BASE`) at all.
Why so? We probably just need tests for both in here.

Reply:
SubObjects are used in many many other tests and test classes, dont think we need them here
In this test, the generic features of the EmfForm were tested, not specifically SubObjects
Problem is, this test is now scoped to a specific test objekt type, before it was `EmfTestSubObject`, now you have to define it when extending `AbstractEmfFormTest`
It was confusing before, its also not easy to explain in text form

Reply:
Where else do we test the UI part of the creation of sub objects?

Btw, note that the generification of `AbstractEmfFormTest` is actually not necessary, so it should not keep us from expanding test coverage.

Consider this, applied on top of the current PR:
```
diff --git a/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java b/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java
index 6c9fc184..04ed4a8c 100644
--- a/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java
+++ b/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java
@@ -9,27 +9,32 @@ import com.softicar.platform.emf.test.simple.EmfTestObject;
 import com.softicar.platform.emf.test.simple.scoped.EmfScopedTestObject;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Objects;
 
-public abstract class AbstractEmfFormTest<R extends IEmfTableRow<R, ?>> extends AbstractEmfTest {
+public abstract class AbstractEmfFormTest extends AbstractEmfTest {
 
-       protected final EmfFormTestFrame<R> frame;
        protected final Collection<Object> creationCallbacks;
-       protected EmfForm<R> form;
+       protected EmfForm<? extends IEmfTableRow<?, ?>> form;
+       protected EmfFormTestFrame<?> frame;
 
        public AbstractEmfFormTest() {
 
-               this.frame = new EmfFormTestFrame<>();
diff --git a/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java b/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java
index 6c9fc184..04ed4a8c 100644
--- a/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java
+++ b/platform-emf/src/test/java/com/softicar/platform/emf/form/AbstractEmfFormTest.java
@@ -9,27 +9,32 @@ import com.softicar.platform.emf.test.simple.EmfTestObject;
 import com.softicar.platform.emf.test.simple.scoped.EmfScopedTestObject;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Objects;
 
-public abstract class AbstractEmfFormTest<R extends IEmfTableRow<R, ?>> extends AbstractEmfTest {
+public abstract class AbstractEmfFormTest extends AbstractEmfTest {
 
-       protected final EmfFormTestFrame<R> frame;
        protected final Collection<Object> creationCallbacks;
-       protected EmfForm<R> form;
+       protected EmfForm<? extends IEmfTableRow<?, ?>> form;
+       protected EmfFormTestFrame<?> frame;
 
        public AbstractEmfFormTest() {
 
-               this.frame = new EmfFormTestFrame<>();
+               this.frame = null;
                this.creationCallbacks = new ArrayList<>();
 
-               setNodeSupplier(() -> frame);
+               setNodeSupplier(() -> Objects.requireNonNull(frame, "Frame was not initialized."));
        }
 
-       protected EmfForm<R> appendEntityForm(R entity) {
+       protected <R extends IEmfTableRow<R, ?>> EmfForm<R> appendEntityForm(R entity) {
 
-               return frame.appendChild(new EmfForm<>(frame, entity));
+               var frame = new EmfFormTestFrame<R>();
+               this.frame = frame;
+
+               EmfForm<R> form = new EmfForm<>(frame, entity);
+               return frame.appendChild(form);
        }
 
-       protected R showForm(R entity) {
+       protected <R extends IEmfTableRow<R, ?>> R showForm(R entity) {
 
                this.form = appendEntityForm(entity);
                form.setCallbackAfterCreation(creationCallbacks::add);
diff --git a/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormRefreshTest.java b/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormRefreshTest.java
index 633ae4f8..d8aa8092 100644
--- a/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormRefreshTest.java
+++ b/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormRefreshTest.java
@@ -5,7 +5,7 @@ import com.softicar.platform.emf.test.EmfTableRowCacheMapTestMethods;
 import com.softicar.platform.emf.test.EmfTestSubObject;
 import org.junit.Test;
 
-public class EmfFormRefreshTest extends AbstractEmfFormTest<EmfTestSubObject> {
+public class EmfFormRefreshTest extends AbstractEmfFormTest {
 
        private final EmfTestSubObject testEntity;
 
diff --git a/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormTest.java b/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormTest.java
index 0e8b7293..ee5ab9f6 100644
--- a/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormTest.java
+++ b/platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormTest.java
```